cmake_modules/Commons.cmake

```diff
@@ -20,11 +20,11 @@ macro(set_common_cxx_flags_for_redex)
             set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++")
         endif ()
 
-        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall")
+        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall -pthread")
         set(COMMON_CXX_FLAGS_NODBG, "-O3")
-        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${COMMON_CXX_FLAGS_NODBG}")
-        set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${COMMON_CXX_FLAGS_NODBG}")
-        set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} ${COMMON_CXX_FLAGS_NODBG}")
+        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${COMMON_CXX_FLAGS_NODBG} -pthread")
+        set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${COMMON_CXX_FLAGS_NODBG} -pthread")
+        set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} ${COMMON_CXX_FLAGS_NODBG} -pthread")
```

Comment:
nit: why repeat this three times when we could put it on `COMMON_CXX_FLAGS_NODBG`once?

Reply:
It still gave me the Missing DSO error.

Reply:
This was what I did
`set(COMMON_CXX_FLAGS_NODBG, "-O3 -pthread")`
but still I get the following error
`libpthread.so.0: error adding symbols: DSO missing from command line`

Reply:
Huh. Okay, we'll keep it like this then because it works. Thanks for the quick response.